### PR TITLE
Add "supports_attachments" field to EmailTemplate model

### PR DESCRIPTION
**Pull request by hugorodgerbrown**
This PR adds a boolean attribute to the `EmailTemplate` model to indicate that the email should not have attachments added. If set to `False` (default=`True`) then the `create_message` method will fail if the calling code tries to pass an attachments kwarg:

```python
>>> template.supports_attachments = False
>>> template.create_message({}, attachments=[...])
ValueError: Email template does not support attachments.
```

This is designed to allow us to hard fail in the event that we do not want attachments.

### CODE REVIEW

**Comment by linusg** [comment]
`black`ification should probably be done at least in a separate commit to make reviewing this easier.

File: appmail/models.py [hunk 1]
```diff
@@ -240,28 +250,36 @@ def create_message(self, context, **email_kwargs):
...
+        if "subject" in email_kwargs:
+            raise ValueError(_("Invalid argument: 'subject' is set from the template."))
+        if "body" in email_kwargs:
+            raise ValueError(_("Invalid argument: 'body' is set from the template."))
+        if "alternatives" in email_kwargs:
+            raise ValueError(
+                _("Invalid argument: 'alternatives' is set from the template.")
+            )
+        if email_kwargs.get("attachments", None) and not self.supports_attachments:
+            raise ValueError(_("Email template does not support attachments."))
```
Comment (hugorodgerbrown) [inline appmail/models.py]: This is the content of the PR - everything else (beyond the definition of the field itself) is Black / isort / prettier.

**Comment by hugorodgerbrown** [comment]
> `black`ification should probably be done at least in a separate commit to make reviewing this easier.

It should - I didn't really expect anyone to review it ;-)

I have highlighted the two-lines that are functional.

File: appmail/tests/test_models.py [hunk 2]
```diff
@@ -161,69 +161,101 @@ def test_create_message(self):
...
+        template = EmailTemplate(
+            subject="Welcome {{ first_name }}",
+            body_text="Hello {{ first_name }}",
+            body_html="<h1>Hello {{ first_name }}</h1>",
+            supports_attachments=False,
+        )
+        with self.assertRaisesMessage(
+            ValueError, "Email template does not support attachments."
+        ):
+            template.create_message({}, attachments=[mock.Mock(spec=MIMEImage)])
```
Comment (hugorodgerbrown) [inline appmail/tests/test_models.py]: This is the functional element in use - if you try to add attachments to a template that doesn't support them, it'll error.

**Pull request merged**